contrib/kubernetes/README.md

```diff
@@ -0,0 +1,214 @@
+# Managing an etcd cluster with Kubernetes
+
+This project includes guidance and sample files to use Kubernetes for managing and monitoring an etcd cluster. Persistent storage backs each pod as it is created, destroyed, and recreated, in order to provide the reliable state required by etcd. This example constructs an etcd cluster of three nodes, but can be extrapolated to any cluster size.
+
+## Architecture
+
+In order for an etcd cluster to work without any discovery we need to have static IP addresses that are well known for each server instance.  We accomplish that with creating individual kubernetes services for each replication controller/pod in our cluster.  Using kubernetes DNS or environment variables lets each pod's etcd know how to connect to the servers in the cluster.  Each of the N replication controller/pod instances will define a label that specifies `app=etcd-N` and each of the N services will select on `app=etcd-N`.
+
+In order to provide a consistent endpoint we also provide a global service that selects all of the replication controller/pod instances that make up our cluster and this is what clients should use.
+
+In order to provide a consistent environment, we create N persistent volumes for etcd to store it's data with N persistent volumes claims.  Therefore, even if a pod is removed, on recreation, it will still have its data directory available and will come up cleanly.  This will only work if the environment can provide that volume on the different nodes, if for instance the persistent volume is node local storage and the node goes down, it wont help.
```

Comment:
I would rather emphasize this (remote persistent volumes like EBS or GCE disk) is actually a prerequisite for this approach/architecture works and to be resilient for pod failure.